extensions/wikia/Search/js/WikiaSearch.js

```diff
@@ -33,20 +35,14 @@ require(['search-tracking', 'uuid', 'wikia.trackingOptIn'], function(searchTrack
 		}.bind(this);
 	};
 
-	var getCurrentScope = function () {
-		return $('#search-v2-scope').val();
-	};
-
 	var getUniqueSearchId = function() {
 		if (this.searchUID) {
 			return this.searchUID;
 		}
 
 		var queryParams = new URL(window.location).searchParams;
-		var searchUID = queryParams.get('searchUID') || uuid();
-		this.searchUID = searchUID;
 
-		return searchUID;
+		return queryParams.get('searchUID') || uuid();
```

Comment:
If searchUID is not available shouldn't you generate it once and assingn to `this.SearchUID`?

Reply:
Good catch - it's not generated backend side, so for first search page id for impression and click tracking is always different 🤔
This is old code, so it could be broken for a long time